internal/ssh/git_ssh.go

```diff
@@ -37,7 +39,9 @@ func runGitCommand(ch ssh.Channel, gitCmd string, key string, ip string) error {
 		return errors.New("gist not found")
 	}
 
-	requireLogin, err := db.GetSetting(db.SettingRequireLogin)
+	// FIXME: this seems to not actually work for clones, the auth process
```

Comment:
Hi @lf- , sorry for the long delay

About this FIXME, does the bug occur when you enable RequireLogin and try to `git clone` another user gist ? Otherwise i'm not sure to encounter the bug

Reply:
Unrelated to this PR, but maybe related to the bug you're encountering, there is a wrong interaction with RequireLogin, when it's true, you can only pull/clone your own gists... which needs to be fixed ofc

Reply:
See #280 if this is the bug you're encountering

Reply:
Or maybe if you encountered this (i try to understand the fixme)

```
5:03AM ERR SSH: Failed to handshake error="[ssh: no auth passed yet, unknown public key]"
```

That means the ssh key git uses has not been added for your user into your opengist instance

Reply:
IIRC it doesn't even reach this code at all. I kind of forget the reproducer as well.

Reply:
I would assume it was the second bug i described..